### ISSUE TITLE
[git][sdl2][gcc 11.1] Build process from commit 64a5e80 is broken in mouse.cpp:599:20:error: ‘bool GFX_IsFullscreen()’ was declared ‘extern’ and later ‘static’ [-fpermissive]

### ISSUE THREAD

**Issue by FredBezies** [joncampbell123/dosbox-x#2620]
**Describe the bug**
I try to build this morning on my archlinux Dosbox-X based on commit 64a5e80.

Build process goes on for some times and then stops.

Here is the start of the error log I get:

```
mouse.cpp:599:20: error: ‘bool GFX_IsFullscreen()’ was declared ‘extern’ and later ‘static’ [-fpermissive]
  599 | static inline bool GFX_IsFullscreen(void) {
      |                    ^~~~~~~~~~~~~~~~
In file included from ../../include/../src/gui/render_scalers.h:23,
                 from ../../include/render.h:28,
                 from mouse.cpp:40:
../../include/video.h:74:6: note: previous declaration of ‘bool GFX_IsFullscreen()’
   74 | bool GFX_IsFullscreen(void);
      |      ^~~~~~~~~~~~~~~~
mouse.cpp: In function ‘const char* Mouse_GetSelected(int, int, int, int, int, int, uint16_t*)’:
mouse.cpp:776:7: warning: this ‘else’ clause does not guard... [-Wmisleading-indentation]
  776 |     } else
      |       ^~~~
mouse.cpp:809:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘else’
  809 |         text[len]=0;
      |         ^~~~
mouse.cpp: In function ‘void Mouse_Select(int, int, int, int, int, int, bool)’:
mouse.cpp:817:13: warning: unused variable ‘page’ [-Wunused-variable]
  817 |     uint8_t page = real_readb(BIOSMEM_SEG,BIOSMEM_CURRENT_PAGE);
      |             ^~~~
```

**To Reproduce**
1. Grab source code commit 64a5e80
2. Use the following configure options: --enable-core-inline --enable-debug --enable-avcodec --prefix=/usr --enable-sdl2
3. Wait a little

**Expected behavior**
Build process going on.

**Environment (please complete the following information):**
 - Archlinux
 - commit SHA1: 64a5e80
 - gcc 11.1.0

**Comment by FredBezies** [comment]
Adding full build log.
[dosbox-x-sdl2-git-1:0.83.14.r230.g64a5e801b-1-x86_64-build.log](https://github.com/joncampbell123/dosbox-x/files/6719883/dosbox-x-sdl2-git-1.0.83.14.r230.g64a5e801b-1-x86_64-build.log)

**Comment by FredBezies** [comment]
Could be a side effect of commit a0e3abf because this is the last commit modifying mouse.cpp

**Comment by rderooy** [comment]
I can confirm the breakage. Only effects the SDL2 build, SDL1 builds fine.

**Comment by FredBezies** [comment]
Fixed by commit 7d568f7.

Thanks a lot. Closing this bug.
